modules/features2d/src/fast.cpp

```diff
@@ -417,6 +418,65 @@ static bool openvx_FAST(InputArray _img, std::vector<KeyPoint>& keypoints,
 
 #endif
 
+namespace hal
+{
+inline int FAST(cv::Mat& src, std::vector<KeyPoint>& keypoints, int threshold, bool nonmax_suppression, int type)
```

Comment:
It will be better to remove this function from the `hal::` namespace, we use it for HAL public interface (defined in [hal.hpp](https://github.com/opencv/opencv/blob/master/modules/imgproc/include/opencv2/imgproc/hal/hal.hpp)).

You can mark it `static` and rename to something like `hal_FAST`.